core/service.go

```diff
@@ -91,10 +112,33 @@ func (srv *RelayService) Serve(ctx context.Context) error {
 		logger.Error("failed to get unrelayed acknowledgements", err)
 		return err
 	}
-	if err := srv.st.RelayAcknowledgements(srv.src, srv.dst, aseqs, srv.sh); err != nil {
-		logger.Error("failed to relay acknowledgements", err)
-		return err
+	if srv.optimizeRelay(*pseqs) {
+		if err := srv.st.RelayAcknowledgements(srv.src, srv.dst, aseqs, srv.sh); err != nil {
+			logger.Error("failed to relay acknowledgements", err)
+			return err
+		}
 	}
 
 	return nil
 }
+
+func (srv *RelayService) optimizeRelay(pseqs RelayPackets) bool {
+	// packet count
+	srcPacketCount := len(pseqs.Src)
+	dstPacketCount := len(pseqs.Dst)
+	if int64(srcPacketCount) >= srv.optimizeCount || int64(dstPacketCount) >= srv.optimizeCount {
+		return true
+	}
+	// time interval
+	if time.Since(srv.srcCurrentTime) >= srv.optimizeInterval || time.Since(srv.dstCurrentTime) >= srv.optimizeInterval {
+		return true
+	}
+	// set current time
+	if srcPacketCount > 0 {
+		srv.srcCurrentTime = time.Now()
+	}
+	if dstPacketCount > 0 {
+		srv.dstCurrentTime = time.Now()
+	}
+	return false
```

Comment:
```
// time interval
if time.Since(srv.srcCurrentTime) >= srv.optimizeInterval || time.Since(srv.dstCurrentTime) >= srv.optimizeInterval {
	return true
}
```
If either condition is met, relay is executed

Reply:
https://github.com/hyperledger-labs/yui-relayer/pull/108/commits/2fd387c3442472d547663d3da0c12ddd4a40d2db